### PR TITLE
[#427] Add not enough disk space screen

### LINKED ISSUE #427: Warn the user about not enough available free space
Syncing takes a significant amount of space. We are working towards fixing that. Until then, we want to let users know that we don't consider there's enough space to continue syncing.
The goal is to not let the wallet paint itself to a corner and be proactive towards detecting this denial-of-service situation where the wallet occupies space that can't be claimed back.

Scenario:
When the app goes to foreground it will attempt to sync. Before doing so, the wallet should check whether there's enough space to continue safely.

use case A: there's enough space. the wallet proceeds to start syncing.

use case B: theres is **not** enough space. The wallet should not proceed to sync.  Instead it should show a warning as the design below.

Preconditions: 
- this free space check happens only after the users has created a new wallet and backed up the keys successfully or restored from backup. 

implementation details:
- the amount of space required should be easily modifiable on code. 
- create mocks to trigger this scenario
- add snapshot test with the corresponding screenshot.
- implement light and dark mode.

### PR DESCRIPTION
**Pull request by Chlup**
Closes #427 

- Logic that works with free disk space is in `DiskSpaceChecker`. Use `WrappedDiskSpaceChecker` to get live or mocked instances of `DiskSpaceChecker`.
- You can either use `mockEmptyDisk` which mocks situation when there is enough space on disk.
- Or you can use `mockFullDisk` to mock situation when there isn't enough space on disk.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

**Review by pacu — APPROVED** [review]
Even though there are open questions on the feature itself, I consider this complete in terms of the defined task.

Let's see why CI is failing

File: secant/Features/NotEnoughFreeSpace/NotEnoughFreeSpaceView.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//
+//  Created by Michal Fousek on 28.09.2022.
+//
+
+import Foundation
+import SwiftUI
+import ComposableArchitecture
+
+struct NotEnoughFreeSpaceView: View {
+    @Environment(\.scenePhase) var scenePhase
```
Comment (LukasKorba) [inline secant/Features/NotEnoughFreeSpace/NotEnoughFreeSpaceView.swift:13]: I think you don't use this anymore, please remove

File: secantTests/HomeTests/HomeTests.swift [hunk 2]
```diff
@@ -209,13 +214,52 @@ class HomeTests: XCTestCase {
...
+        store.receive(.updateRoute(nil))
         store.receive(.synchronizerStateChanged(.unknown))
         store.receive(.updateSynchronizerStatus)
 
         // long-living (cancelable) effects need to be properly canceled.
         // the .onDisappear action cancles the observer of the synchronizer status change.
         store.send(.onDisappear)
     }
+
+    func testOnAppear_notEnoughSpaceOnDisk() throws {
```
Comment (LukasKorba) [inline secantTests/HomeTests/HomeTests.swift:226]: Do we miss testOnAppear_enoughSpaceOnDisk? It should go the other way and register for the synchronizer's status update and spin up the whole wallet.

File: secant/Wrappers/WrappedDiskSpaceChecker.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        )
+    }()
+
+    static let mockFullDisk = {
+        return WrappedDiskSpaceChecker(
+            freeSpaceRequiredForSync: { 1024 },
+            hasEnoughFreeSpaceForSync: { false },
+            freeSpace: { 0 }
+        )
+    }()
```
Comment (LukasKorba) [inline secant/Wrappers/WrappedDiskSpaceChecker.swift:40]: [non-blocking] These two could be simplified a bit
```swift
static let mockEmptyDisk = WrappedDiskSpaceChecker(
            freeSpaceRequiredForSync: { 1024 },
            hasEnoughFreeSpaceForSync: { true },
            freeSpace: { 2048 }
        )
```

**Review by LukasKorba — CHANGES_REQUESTED** [review]
Great work, a few small comments

File: secant/Wrappers/WrappedDiskSpaceChecker.swift (hunk 3 above)
Comment (LukasKorba) [inline secant/Wrappers/WrappedDiskSpaceChecker.swift:40]: Bitrise is failing
```
cannot infer return type for closure with multiple statements; add explicit type to disambiguate
    static let live = {
```
so we need to at least
```swift
static let live: WrappedDiskSpaceChecker = {
```

File: secantTests/HomeTests/HomeTests.swift (hunk 2 above)
Comment (Chlup) [reply secantTests/HomeTests/HomeTests.swift:226]: `testOnAppear` does exactly this. It was already there and I just mocked empty disk for it.

**Pull request merged**